benchmark/src/androidTest/java/com/example/benchmark/AutoBoxingBenchmark.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2019 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.benchmark
+
+import androidx.benchmark.BenchmarkRule
+import androidx.test.filters.LargeTest
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@LargeTest
+@RunWith(JUnit4::class)
+class AutoBoxingBenchmark {
```

Comment:
I did some experimenting with this one locally - not really happy with it yet. Let's remove this file for now.

If I messed around with the structure, it seemed I could get the VM to reuse boxed integers and make performance very similar (or worse, have one run pollute the next), e.g.:

    @Test
    fun integerArtCacheAlloc() {
        benchmarkRule.measure {
            (0..99).map { Integer(it) }
        }
    }

    @Test
    fun integerAlloc() {
        benchmarkRule.measure {
            (1000..1099).map { Integer(it) }
        }
    }

Reply:
Isn't the point to write code in a way that doesn't take advantage of this and show that it doesn't though?

Reply:
What if I added this test case in here and named it something like `integerJvmReuseAlloc`?